### PR TITLE
Data store

### PR DESCRIPTION
**Pull request by jyan1999**
# Description

Changed the game UI and updated docker configurations to allow for data collection. 

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/overcooked_demo/server/static/js/index.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ $(function() {
         params.layouts = [params.layout]
         data = {
             "params" : params,
-            "game_name" : "overcooked",
+            //changed to psiturk to always enable data collection
+            //will decide whether to write data during class creation
+            "game_name" : "psiturk",
```
Comment (micahcarroll) [inline src/overcooked_demo/server/static/js/index.js:16]: We probably should remove all mentions of psiturk from the code

File: src/overcooked_demo/server/static/templates/index.html [hunk 2]
```diff
@@ -64,6 +64,8 @@
     <input type="checkbox" id="showPotential" name="showPotential">
     <label for="Use Old Dynamics">Old dynamics?</label>
     <input type="checkbox" id="oldDynamics" name="oldDynamics">
```
Comment (micahcarroll) [inline src/overcooked_demo/server/static/templates/index.html:66]: I would set this to true by default. Or actually, maybe we should have the checkbox be `use v2 dynamics` and change the related documentation in the repo. I think the standard should probably just be to use the old dynamics

**Review by micahcarroll — COMMENTED** [review]
Leaving a first round of comments! Looks reasonable so far!

File: src/overcooked_demo/README.md [hunk 3]
```diff
@@ -64,8 +64,11 @@ If a more complex or custom loading routing is necessary, one can subclass the `
 
 ### 1): Using Docker Compose CLI to deploy on EC2
 With the Overcooked demo, you can test the interaction between two human players. To do this, you need to deploy this code on a server (https://docs.docker.com/language/python/deploy/). 
-After successful deployment, the first user should open http://[server_ip_address]/, select the human keyboard input for both players and click on "Create game". If everything has been successful, he will receive a message: "Waiting for game to start".
-Another user should open a page at http://[server_ip_address]/psiturk to start the game.  
+After successful deployment, the first user (User 1) should go to http://[server_ip_address]/, select the human keyboard input for both players, and select the setting for this game (e.g. whether to collect data or to run the game with old dynamics, which means the soup starts cooking automatically when all ingredients are received), and click "Create game". If everything has been successful, he will receive a message: "Waiting for game to start".
```
Comment (micahcarroll) [inline src/overcooked_demo/README.md:67]:
```suggestion
After successful deployment, the first user (User 1) should go to http://[server_ip_address]/, select the human keyboard input for both players, and select the setting for this game (e.g. whether to collect data and/or to run the game with new dynamics, which means the soup requires an additional "interact" action to start cooking), and click "Create game". If everything has been successful, he will receive a message: "Waiting for game to start". 
```

File: src/overcooked_demo/README.md [hunk 4]
```diff
@@ -93,6 +96,16 @@ overcooked-demo-up
 ```
 
 7: Once the server is up, you can access the game through your browser by passing in the public ip of your server and port 80.
+
+### 3): Running predefined experiments
+
+One can also define a series of experiments to be run when players join the game. To do that, go to [`config.json`](./server/config.json) and modify the "predefined" parameters by changing what the layouts you want to include and the settings for them. 
+
+Save the changes and spin up a server by ways described in the earlier sections. User 1 should go to http://[server_ip_address]/predefined, and he/she should be placed in a waiting room, and he will see how many layouts are there in the current series of experiments. User 2 can now go to http://[server_ip_address]/predefined, and the game should automatically starts. The games will be run back to back until it goes through all the predefined layouts in the config file.
```
Comment (micahcarroll) [inline src/overcooked_demo/README.md:104]:
```suggestion
Save the changes and spin up a server by ways described in the earlier sections. User 1 should go to http://[server_ip_address]/predefined, where they should be placed in a waiting room, and they will see how many layouts are there in the current series of experiments. User 2 can now go to http://[server_ip_address]/predefined, and the game should automatically start. The games will be run back to back until it goes through all the predefined layouts in the config file.
```

File: src/overcooked_demo/server/config.json [hunk 5]
```diff
@@ -5,13 +5,14 @@
     "MAX_GAME_LENGTH" : 120,
     "AGENT_DIR" : "./static/assets/agents",
     "MAX_FPS" : 30,
-    "psiturk" : {
+    "predefined" : {
         "experimentParams" : {
             "layouts" : ["counter_circuit", "cramped_room"],
-            "gameTime" : 11,
+            "gameTime" : 5,
```
Comment (micahcarroll) [inline src/overcooked_demo/server/config.json:11]: Is this 5 seconds? I'm guessing that's fine given that this file is just for demonstration purposes, right?

**Review by micahcarroll — APPROVED** [review]
LGTM, with minor comments!

File: src/overcooked_demo/server/config.json (hunk 5 above)
Comment (jyan1999) [reply src/overcooked_demo/server/config.json:11]: Yeah it is 5 seconds. This was mostly so I could test the behaviors in less time during development. I don't know how it was designed to be used, but currently this only acts as a demo for how you can run predefined experiments

**Pull request merged**